packages/grid/_modules_/grid/hooks/features/rows/useGridRowsHydration.ts

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { GridRowsHydrationApi } from '../../../models/api/gridRowsHydrationApi';
+import { DataGridProcessedProps } from '../../../models/props/DataGridProps';
+import { getCurrentPageRows } from '../../utils/useCurrentPageRows';
+import { useGridApiMethod } from '../../utils/useGridApiMethod';
+import { GridRowId } from '../../../models/gridRows';
+import { useGridSelector } from '../../utils/useGridSelector';
+import {
+  gridDensityRowHeightSelector,
+  gridDensityFactorSelector,
+} from '../density/densitySelector';
+import { gridFilterStateSelector } from '../filter/gridFilterSelector';
+import { gridPaginationSelector } from '../pagination/gridPaginationSelector';
+import { gridSortingStateSelector } from '../sorting/gridSortingSelector';
+
+/**
+ * @requires useGridPageSize (method)
+ * @requires useGridPage (method)
+ */
+export const useGridRowsHydration = (
```

Comment:
I try to have 1 feature hook = 1 sub state to simplify the codebase understanding.

Could we either

1. Move `state.rows.meta` into some `state.rowDimensions` for instance.
2. Call `useGridRowsHydratation` inside `useGridRows`. Which seems to require more changes due to the dependances of this hook, so maybe not worth it.

Reply:
In general, I agree with the approach of having 1 feature hook and 1 substate. In this case tho it is a bit different because the row dimensions do make sense to the part of the row state as you can't separate them (same logic as for the columns). Ideally, the `useGridRowsHydration` should be part of the `useGridRows` (as it was in the beginning) but then there is a problem because of the pagination.

I would suggest keeping it as is for now and maybe reviewing it already took more time than planned for this feature to be done.

Reply:
The only problem in doing it later is that, as long as we export selectors like `gridRowsStateSelector` (which I would be in favor of dropping), the `state.rows` is public API so we won't be able to change it before v6.

Reply:
I moved the Meta state outside of the rows state and renamed the files accordingly.